fixedSize.lua

```diff
@@ -30,56 +32,78 @@ fixedSizeTA.bake = function(self)
     local width, height = self.width, self.height
     local widthPadded = width + self.spacing + self.extrude * 2 + self.padding * 2
     local heightPadded = height + self.spacing + self.extrude * 2 + self.padding * 2
-    
+    local maxIndex = self.imagesSize
+    local data
+
     local widthCanvas = columns * widthPadded
     if widthCanvas > self._maxCanvasSize then
       columns = floor(self._maxCanvasSize / width)
       widthCanvas = columns * widthPadded
     end
-    
+
     local rows = ceil(self.imagesSize / columns)
     local heightCanvas = rows * heightPadded
     if heightPadded > self._maxCanvasSize then
       rows = floor(self._maxCanvasSize / height)
       heightCanvas = rows * heightPadded
     end
-    
-    if columns * rows < self.imagesSize then
-      error("Cannot support "..tostring(self.imagesSize).." images, due to system limits of canvas size. Max allowed on this system: "..tostring(columns * rows))
-    end
-    
-    local extrudeQuad = lg.newQuad(-self.extrude, -self.extrude, width+self.extrude*2, height+self.extrude*2, self.width, self.height)
+
     widthCanvas, heightCanvas = widthCanvas - self.spacing, heightCanvas - self.spacing
     if self.bakeAsPow2 then
       widthCanvas = math.pow(2, math.ceil(math.log(widthCanvas)/math.log(2)))
       heightCanvas = math.pow(2, math.ceil(math.log(heightCanvas)/math.log(2)))
     end
-    local canvas = lg.newCanvas(widthCanvas, heightCanvas, self._canvasSettings)
-    local maxIndex = self.imagesSize
-    lg.push("all")
-    lg.setBlendMode("replace")
-    lg.setCanvas(canvas)
-    for x=0, columns-1 do
-      for y=0, rows-1 do
-        local index = (x+y*rows)+1
-        if index > maxIndex then
-          break
+
+    if self._pureImageMode then
+      data = newImageData(widthCanvas, heightCanvas, "rgba8")
+      for x=0, columns-1 do
+        for y=0, rows-1 do
+          local index = (x+y*rows)+1
+          if index > maxIndex then
+            break
+          end
+          local x, y = x * widthPadded + self.padding + self.extrude, y * heightPadded + self.padding + self.extrude
+          local image = self.images[index]
+          data:paste(image, x, y, 0, 0, image:getDimensions())
```

Comment:
image is a table containing the image, it should be `image.image` and `image.image:getDimensions()`